packages/disploy/src/structs/CommandInteraction.ts

```diff
@@ -53,6 +54,13 @@ export class CommandInteraction extends BaseInteraction {
 		});
 	}
 
+	public showModal(modal: Modal) {
```

Comment:
Maybe we should make a generic that takes in a structure and turns it into a union with the structure and unknown, then check if it's a JSON or a structure and add .toJSON() if necessary.

Reply:
something like

```ts
export type DiscordPayload<T> = T | unknown;

public showModal(modal: DiscordPayload<Modal>)
```

Reply:
Check if 4621b26484ec3e2eda15d7818cc876e5dd44b755 satisfies requested changes.